### PR TITLE
Remove inet_ntop definitions for mingw

### PR DESCRIPTION
**Pull request by devlexanderxyz**
The newer versions of MinGW already have these functions defined in their respective headers.
This causes issues when compiling with CGo, or Kotlin Native, as they use MinGW behind the scenes.
The fix is to remove this piece of code, and anyone using an older version of MinGW will either have to redefine these functions or update.

### CODE REVIEW

**Comment by nxrighthere** [comment]
To fix this properly, [these](https://github.com/nxrighthere/ENet-CSharp/blob/master/Source/Native/CMakeLists.txt#L8) definitions should be added to enable compilation with previous versions.

File: include/enet.h [hunk 1]
```diff
@@ -5358,188 +5358,6 @@ extern "C" {
 
     #ifdef _WIN32
 
-    #ifdef __MINGW32__
```
Comment (zaklaus) [inline include/enet.h:5361]: Let's enable compat mode by having:
```c
#if defined(__MINGW32__) && defined(ENET_MINGW_COMPAT)
```

Comment (zaklaus) [reply include/enet.h:5361]: If there are any suggestions, I'll be happy to discuss them.

**Review by zaklaus — CHANGES_REQUESTED** [review]

File: include/enet.h (hunk 1 above)
Comment (devlexanderxyz) [reply include/enet.h:5361]: Yeah, I think that would be great!

**Review by devlexanderxyz — COMMENTED** [review]
I've added the changes in the latest commit.

**Review by zaklaus — APPROVED** [review]
Thanks for your contribution!

**Pull request merged**